src/serac/physics/tests/thermal.cpp

```diff
@@ -155,10 +154,10 @@ void functional_test_dynamic(double expected_temp_norm)
   EXPECT_NEAR(expected_temp_norm, norm(thermal_solver.temperature()), 1.0e-6);
 }
 
-TEST(Thermal, 2DLinearStatic) { functional_test_static<1, 2>(2.2909240); }
-TEST(Thermal, 2DQuadStatic) { functional_test_static<2, 2>(2.29424403); }
-TEST(Thermal, 3DLinearStatic) { functional_test_static<1, 3>(46.6285642); }
-TEST(Thermal, 3DQuadStatic) { functional_test_static<2, 3>(46.6648538); }
+TEST(Thermal, 2DLinearStatic) { functional_test_static<1, 2>(3.0184961962491594); }
+TEST(Thermal, 2DQuadStatic) { functional_test_static<2, 2>(3.0186175426112518); }
+TEST(Thermal, 3DLinearStatic) { functional_test_static<1, 3>(4.0564914381870958); }
+TEST(Thermal, 3DQuadStatic) { functional_test_static<2, 3>(4.0573746914550073); }
```

Comment:
I don't want to be too disagreeable, but I did say that I was not going to approve PRs that introduce new tests with magic number answers based on the norm of the solution. Can we update this test to:

1. use one of the patch test meshes[^1].
2. use an analytic solution
3. demand that the relative error between the exact and approximate solutions is small.

When all is said and done, it should only amount to a difference of 5-10 lines of code, but it improves the value of the test significantly.

[^1]: In general, I don't see a compelling reason to use the beam/star meshes for any of the correctness tests, since they contain symmetries that make it possible for the finite element machinery to be wrong and still get the right answer by accident.

Reply:
I was thinking that would be a separate PR, but I can take a stab at it here.

Reply:
oops, [9ad6528](https://github.com/LLNL/serac/pull/840/commits/9ad6528f1daf46b41fd5261ececf8dcd8f6d87f2) was a bigger change than I envisioned, but that is even more thorough!

Reply:
Thanks go to @btalamini for making is patch test infrastructure so adaptable! I'm getting an error in the flux case that I need to investigate.